src/pages/admin/donate/components/generic-form/GenericForm.tsx

```diff
@@ -301,7 +301,10 @@ export function createGenericForm<T extends { id?: number }>(fields: GenericForm
                     if (!f.isRequired) return false;
                     const value = formState[f.name];
                     if (Array.isArray(value)) return value.length === 0;
-                    return !String(value ?? '').trim();
+                    const normalized = String(value ?? '')
+                        .trim()
+                        .replace(/\s+/g, ' ');
+                    return !normalized;
```

Comment:
also move to `utils`. Moreover you can cover both functions with additional unit tests now :)

```suggestion
                    export const getNormalizeText = (value: unknown): string =>
                        String(value ?? '')
                            .trim()
                            .replace(/\s+/g, ' ');
```

Reply:
Added new commits addressing all comments. Please take a look